docs/notebooks/asynchronous_bayesian_optimization.pct.py

```diff
@@ -0,0 +1,373 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.12.0
+#   kernelspec:
+#     display_name: 'Python 3.7.5 64-bit (''.venv'': venv)'
+#     name: python3
+# ---
+
+# %% [markdown]
+# # Asynchronous Bayesian optimization with Trieste
+#
+# In this notebook we show how Bayesian optimization can be done asynchronuosly. That is pertinent in scenarios when the objective function we are aiming to optimize can be run for several points in parallel, and observations might return back at different times. To avoid wasting resources we immediatelly request the next point asynchronuosly, taking into account points that still being evaluated.
+#
+# To contrast this approach with regular [batch optimization](batch_optimization.ipynb), this notebook also shows how to run parallel synchronous batch approach.
```

Comment:
it would be good to add a sentence on potential performance differences, async one will have more information as some observations would have had arrived and model could be improved, right?

Reply:
I don't know if we can clearly make this call. Although [this paper](https://arxiv.org/abs/1901.10452) claims async is actually more efficient in both runtime and sample efficiency, i wouldn't rush to set user expectations in general. Any ideas how to phrase it without claiming much?

Reply:
I think its enough to put a stress on "potential" and give usual "depending on the use case" etc - I do think its important to say smth about why would you use it - link to that paper would also be good to add